src/main/java/flaxbeard/steamcraft/SteamcraftBook.java

```diff
@@ -433,10 +489,49 @@ public static void registerExosuit() {
             if (Config.enableJumpAssist) {
                 SteamcraftRegistry.addResearch("research.JumpAssist.name", "!research.ExoHeel.name", new BookPageItem("research.JumpAssist.name", "research.JumpAssist.0", true, new ItemStack(SteamcraftItems.jumpAssist)), new BookPageCrafting("", "jumpAssist1", "jumpAssist2"));
             }
+            if (Config.enableAnchorHeels) {
+                if (Config.enableLeadPlate && OreDictionary.getOres("ingotLead").size() > 0 &&
+                  !Config.enableAnchorAnvilRecipe) {
+                    SteamcraftRegistry.addResearch(
+                      "research.AnchorHeels.name",
+                      "!research.ExoHeel.name",
+                      new BookPageItem("research.AnchorHeels.name",
+                        "research.AnchorHeelsLead.0", true,
+                        new ItemStack(SteamcraftItems.anchorHeels)),
+                      new BookPageCrafting("", "anchorHeels")
+                    );
+                } else {
+                    SteamcraftRegistry.addResearch(
+                      "research.AnchorHeels.name",
+                      "!research.ExoHeel.name",
+                      new BookPageItem("research.AnchorHeels.name",
+                        "research.AnchorHeelsIron.0", true,
+                        new ItemStack(SteamcraftItems.anchorHeels)),
+                      new BookPageCrafting("", "anchorHeels")
+                    );
+                }
+            }
             SteamcraftRegistry.addResearch("research.ExoFoot.name", "category.Exosuit.name");
             if (Config.enableDoubleJump) {
                 SteamcraftRegistry.addResearch("research.DoubleJump.name", "!research.ExoFoot.name", new BookPageItem("research.DoubleJump.name", "research.DoubleJump.0", true, new ItemStack(SteamcraftItems.doubleJump)), new BookPageCrafting("", "doubleJump1", "doubleJump2"));
             }
+            if (Config.enableHydrophobic) {
+                SteamcraftRegistry.addResearch(
+                  "research.Hydrophobic.name",
+                  "!research.ExoFoot.name",
+                  new BookPageItem("research.Hydrophobic.name", "research.Hydrophobic.0", true,
+                    new ItemStack(SteamcraftItems.coatingsHydrophobic)),
+                  new BookPageCrafting("", "hydrophobic")
+                );
+            }
+            if (Config.enablePyrophobic) {
+                SteamcraftRegistry.addResearch(
+                  "research.Pyrophobic.name", "!research.ExoFoot.name",
+                  new BookPageItem("research.Pyrophobic.name", "research.Pyrophobic.0", true,
+                    new ItemStack(SteamcraftItems.coatingsPyrophobic)),
+                  new BookPageCrafting("", "pyrophobic")
+                );
+            }
```

Comment:
This applies to the entire deal with Config and research, is there a way you could create a more general function for this? There seems to be a lot of repeated code. Possibly pass in a boolean array and if it passes that register a string and have another argument for the !